### PR TITLE
added SIYI N7 flight controller config

### PR DESCRIPTION
**Pull request by comla-x**
image

### CODE REVIEW

**Comment by dagar** [comment]
At a glance this looks fine, is their any detailed technical information online?

**Comment by dagar** [comment]
For now can you add the board to both github actions and Jenkins so that it's built automatically? We're hoping to automatically build everything in the near future.

https://github.com/PX4/PX4-Autopilot/blob/068b1494fc1329675e4f0d2d558b572337d42656/.github/workflows/compile_nuttx.yml#L21-L22

https://github.com/PX4/PX4-Autopilot/blob/068b1494fc1329675e4f0d2d558b572337d42656/.ci/Jenkinsfile-compile#L36-L38

**Comment by dagar** [comment]
Can you fix the trivial astyle failure? https://github.com/PX4/PX4-Autopilot/actions/runs/5901241739/job/16007054860?pr=21979

image

File: boards/siyi/n7/nuttx-config/include/board_dma_map.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/siyi/n7/nuttx-config/include/board_dma_map.h:3]:
```suggestion
 *   Copyright (c) 2023 PX4 Development Team. All rights reserved.
```

If you don't mind I would do a quick pass to set these to the current year, personally I use it as a reference for when stuff was added.

Comment (comla-x) [reply boards/siyi/n7/nuttx-config/include/board_dma_map.h:3]: > 

Thank you very much for your review. If possible, could you help us correct any incorrect parts? Thank you. Additionally, please let us know which content we need to fix ourselves, and we will make the corrections immediately.

**Comment by comla-x** [comment]
@dagar Additionally, may I ask if we can apply for a dedicated VID (Vendor ID) for SIYI? Currently, we are using the VID and PID (Product ID) of PX4 FMU V6C.

**Comment by comla-x** [comment]
> At a glance this looks fine, is their any detailed technical information online?



> At a glance this looks fine, is their any detailed technical information online?

Could you please browse this link? If any additional information is needed, we will provide it as requested. Thank you.
https://shop.siyi.biz/products/siyi-n7?VariantsId=10788

**Comment by comla-x** [comment]
> At a glance this looks fine, is their any detailed technical information online?

Could you please browse this link? If any additional information is needed, we will provide it as requested. Thank you.
https://shop.siyi.biz/products/siyi-n7?VariantsId=10788

**Comment by vincentpoont2** [comment]
@dagar @mrpollo This flight controller copied the pinout and design of the Holybro Durandal. It undercut the original designer/manufacturer without consent, calling it their own name and design.

We believe that this action should not be encouraged and this product should not be supported.

**Comment by SIYI-TECH** [comment]
Dear Developers,

The product design and user experience of SIYI autopilot will be brand new to customers. The N7 autopilot is going to work as a set with SIYI gimbal cameras, radios, and future ideas of building an ecosystem.

For users, the SIYI design is unique. We’ve compared the pin map in relevant Holybro and CUAV autopilots, they are quite consistent since they are following open-source standards and are from the same generation. The difference is only in some of the sensor models used.

The essential design and the interfaces of the N7 autopilot were carefully decided based on SIYI team's own understanding of the product requirement and following open-source hardware standard. As customers see, the N7 autopilot is quite different.

The purpose that we design N7 flight controller is to contribute to a systematic solution for the PX4 ecosystem including gimbal camera, radio controller, smart battery, and other core parts for a complete system. SIYI has been working on integrating Mavlink protocol in all the core parts and it has been done in gimbal camera and radios.
We sincerely hope PX4 can understand and support the SIYI strategy to empower the open-source ecosystem. We believe that the community will have a deep feeling of SIYI’s contribution very soon, probably in less than 6 months.

**Comment by vincentpoont2** [comment]
This design is not unique, as it copied the Holybro Durandal Pinmap design and code. Copying another company's or someone else's work without permission and calling it their own work should not be encouraged. There are many H7 flight controllers on the market, none of which have blatantly copied another's work and called it their own.

Allowing this will set a bad precedent for the future.

**Comment by cuhome** [comment]
This time I support holybro, because this has violated the BSD3 license. Durandal is maintained by the manufacturer, and anyone who copies it needs to obtain authorization.

**Comment by cuhome** [comment]
@comla-x Since you talked about CUAV, let me tell you, where are you wrong. Both holybro and CUAV have obtained the authorization of pixhawk, and the intellectual property of pixhawk belongs to Dronecode. I can manufacture pixhawk products with small differences under the authorization of Dronecan. You choose to copy the results of holybro, then you should obtain holybro authorization.

**Comment by cuhome** [comment]
@vincentpoont2 Stay tuned for N7 getting Durandal's bootloader in a non-compliant way. They are trying to do that. They are lying, I have never seen such a manufacturer.

**Comment by mrpollo** [comment]
Hey everyone, I have been in touch with SIYI, and they are willing to work with the community to resolve this issue. Sadly GitHub is not the right place to do this; I will contact all affected parties and try to help mediate this issue. I also wanted to thank CUAV and Holybro for raising this concern and for their willingness to work with SIYI to find a resolution. 

Lastly, I wanted to mention that Dronecode can only help to mediate between members and point out that Holybro owns the Durandal design that is openly available on their GitHub page https://github.com/Holybro/Hardware

**Comment by QioTek** [comment]
This PR clearly cloned the Durandal board and removed the contributor/author information. Their PR code similarity reached 99%, only changing the contributor to SIYI, and publicly declare it as their design and maintenance. This behavior violates the principles of the BSD-3 protocol and not respect contributors, which is very interesting. I would like to know if PX4 will encourage such behavior that violates the BSD-3 license ? 

If this behavior is tacitly allowed, do we have the same reason to remove the BSD-3 clause license for PX4 tomorrow.

@mrpollo @dagar Thank you for providing the information to let me know that Durandal hardware is open-source, but what I would like to know more is whether we encourage this behavior of stripping contributor information and claiming to be their work. Thank you.

**Comment by cuhome** [comment]
@mrpollo Please note that siyi is not a dronecode member yet, this is not a mediation between members. First, are they willing to become members? Second, are they willing to modify or obtain holybro authorization?

**Comment by dagar** [comment]
> @mrpollo @dagar Thank you for providing the information to let me know that Durandal hardware is open-source, but what I would like to know more is whether we encourage this behavior of stripping contributor information and claiming to be their work. Thank you.

@QioTek I'm not aware of any contributor information being stripped, if you see this happening please message me directly.

For the vast majority of boards the actual board support is created by first duplicating something similar (eg boards/px4/fmu-v6x) and if everything was already copyright PX4 Development Team I update the year.

**Comment by QioTek** [comment]
> > @mrpollo @dagar Thank you for providing the information to let me know that Durandal hardware is open-source, but what I would like to know more is whether we encourage this behavior of stripping contributor information and claiming to be their work. Thank you.
> 
> @QioTek I'm not aware of any contributor information being stripped, if you see this happening please message me directly.
> 
> For the vast majority of boards the actual board support is created by first duplicating something similar (eg boards/px4/fmu-v6x) and if everything was already copyright PX4 Development Team I update the year.

I think Holybro will tell you that they have enough evidence.

**Comment by mcsauder** [comment]
I personally think it is hilarious that one Chinese company is mad about another Chinese company ripping off an open source design. I also agree this PR is not the place for this discussion, take it up in a place where BSD-3 can be prosecuted, not a place where people try to live by their honor. International patent laws do not apply to schematics that are open source, and schematics are not board layouts.

**Comment by mcsauder** [comment]
I will make one last statement in this thread only because it occurred earlier in this thread and because one of my Chinese colleagues said this to me earlier,  and I will use the previous statements in this this thread to demonstrate my point, he said, "don't trust, they lie."
 Sounds like a previous statement. That is why what is in the open is in the open, it doesn't take a genius to quote sun tzo.

**Pull request merged**